pep/app/pods/read/document/controller.ts

```diff
@@ -407,6 +431,20 @@ export default class ReadDocument extends Controller {
             }
         });
     }
+
+    @action
+    async printDocument() {
+        let url = `${ENV.apiBaseUrl}/${ENV.apiNamespace}/Documents/Downloads/PDF/${this.model.id}/?client-id=${ENV.clientId}`;
+        if (this.session.isAuthenticated && this.session.data.authenticated.SessionId) {
+            url += `&client-session=${this.session.data.authenticated.SessionId}`;
+        }
```

Comment:
you could make the `downloadAuthParams` getter (which is the cleaner way to serialize url params i think) work for this too by just tweaking the getter to conditionally add the `client-session` param to the object only when they are logged in.